src/org.hdfgroup.object/hdf/object/h5/H5ScalarDS.java

```diff
@@ -1869,7 +1910,9 @@ private Object getAttrValue(long oid, String aname) {
                         log.trace("getAttrValue(): id {} is unsigned", atid);
                         avalue = convertFromUnsignedC(avalue, null);
                     }
-                    if (dsDatatype.isReference())
+                    if (dsDatatype.isRef() && (avalue instanceof byte[]))
+                        ((H5ReferenceType)dsDatatype).setData((ArrayList<byte[]>)avalue);
+                    else if (dsDatatype.isReference())
```

Comment:
I'm not really sure what the purpose of having isRef() and isReference() is. Not only is it confusing, but they have the same implementation, with isReference just being an override in H5Datatype of the base Datatype class. I'd suggest removing isReference() since I can't see any differences between the two functions.

Reply:
Again the problem is that new refs cannot read 1.10 refs in a container type.

Reply:
The two functions are exactly the same thing. There shouldn't be any reason to have both of them, regardless of 1.10 vs 1.12 container types. The base Datatype class already had isRef() so that should be usable everywhere.

Reply:
Yep, looks like I just messed up.